Composer/packages/client/src/recoilModel/dispatchers/lu.ts

```diff
@@ -27,37 +29,39 @@ const initialBody = '- ';
  *
  */
 
-const luFilesAtomUpdater = (
+const updateLuFiles = (
+  { set }: CallbackInterface,
+  projectId: string,
   changes: {
     adds?: LuFile[];
     deletes?: LuFile[];
     updates?: LuFile[];
   },
-  filter?: (oldList: LuFile[]) => (changeItem: LuFile) => boolean
+  needUpdate?: (current: LuFile, changed: LuFile) => boolean
 ) => {
-  return (oldList: LuFile[]) => {
-    const updates = changes.updates ? (filter ? changes.updates.filter(filter(oldList)) : changes.updates) : [];
-    const adds = changes.adds ? (filter ? changes.adds.filter(filter(oldList)) : changes.adds) : [];
-    const deletes = changes.deletes
-      ? filter
-        ? changes.deletes.filter(filter(oldList)).map(({ id }) => id)
-        : changes.deletes.map(({ id }) => id)
-      : [];
-
-    // updates
-    let newList = oldList.map((file) => {
-      const changedFile = updates.find(({ id }) => id === file.id);
-      return changedFile ?? file;
-    });
+  const { updates, adds, deletes } = changes;
 
-    // deletes
-    newList = newList.filter((file) => !deletes.includes(file.id));
+  // updates
+  updates?.forEach((luFile) => {
+    set(luFileState({ projectId, luFileId: luFile.id }), (preFile) =>
+      needUpdate ? (needUpdate(preFile, luFile) ? luFile : preFile) : luFile
```

Comment:
??

This looks strange. If needs update, we check to see if needs update. I think we need to rename some variables or add a comment for clarification. Also, avoid nested ternaries because they are hard to read.

Reply:
Ahh I see what's going on here now. I didn't realize `needUpdate` was a function arg. I would suggest providing a default value for this and always invoking it instead of doing this check. OR rename the argument to something that looks like a it is a func. `needUpdate` to me looked like it should be a boolean value.